code/modules/food/cooking_machines.dm

```diff
@@ -195,23 +195,33 @@ var/global/ingredientLimit = 10
 			src.reagents.trans_id_to(target_food, reagent.id, max(5, target_food.w_class * 5) / reagents.reagent_list.len)
 	return
 
+/obj/machinery/cooking/proc/cook_after(var/delay, var/numticks = 5) //adaptation of do_after()
+	var/delayfraction = round(delay/numticks)
+	for (var/i = 1 to numticks)
+		sleep(delayfraction)
+		if (!src.ingredient || !active || get_turf(src.ingredient)!=get_turf(src))
+			return 0
+	return 1
+
 /obj/machinery/cooking/proc/cook(var/foodType)
 	src.active = 1
 	src.icon_state = src.icon_state_on
-	sleep(src.cookTime)
-	if(!src.ingredient || !active) return
+	if (cook_after(src.cookTime, 60))
+		src.makeFood(foodType)
+		playsound(get_turf(src),src.cookSound,100,1)
 	src.active = 0
 	src.icon_state = initial(src.icon_state)
-	playsound(get_turf(src),src.cookSound,100,1)
-	src.makeFood(foodType)
 	return
 
 /obj/machinery/cooking/proc/makeFood(var/foodType)
-	var/obj/item/weapon/reagent_containers/food/new_food = new foodType(src.loc,src.ingredient)
+	var/obj/item/I = src.ingredient
+	var/obj/item/weapon/reagent_containers/food/new_food = new foodType(src.loc,I)
 	if(cooks_in_reagents)
 		transfer_reagents_to_food(new_food)
-	qdel(src.ingredient)
-	src.ingredient = null
+	I.reagents.trans_to(new_food,I.reagents.total_volume)
+	new_food:ingredients += I //<wwjnc> I'd rather have a : than copypasted code
+	new_food:update()		  //<wwjnc> just add a comment so people don't blow up everything
```

Comment:
I knew :::'s are bad and prone to runtiming, but this seemed like a legitimate situation to use one. This implementation is safe because new_food will always be either /food/drinks/bottle/customizable or /food/snacks/customizable. Without using :::'s, I'd have to do two separate istype() to declare new_food, since they're on different pathdirs and their common parent doesn't have a var/ingredients or a proc/update(). Then I'd have to essentially copypaste the code twice. I checked with the IRC first and was given the "eh, sure, whatever", is it really that bad?